File: src/exercicios/05_EstruturasDeDados/04_Vetor/Strings/00-strlen.py
#  -*- coding: utf-8 -*-
#    @package: 00-strlen.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Implemente a função strlen.


def strlen(string):
    '''Retorna a quantidade de caracteres da string.'''

    # Implemente o código aqui.
    pass


if __name__ == '__main__':
    str0 = None
    str1 = 'a'
    str2 = 'Hello World!'
    str3 = 'Olha a pamonha.'
    str4 = 'Um, dois, feijão com arroz...'

    print('strlen(\'{}\') = {}'.format(str0, strlen(str0)))
    print('strlen(\'{}\') = {}'.format(str1, strlen(str1)))
    print('strlen(\'{}\') = {}'.format(str2, strlen(str2)))
    print('strlen(\'{}\') = {}'.format(str3, strlen(str3)))
    print('strlen(\'{}\') = {}'.format(str4, strlen(str4)))

# Resultado esperado:
#
# strlen('(None)') = 0
# strlen('a') = 1
# strlen('Hello World!') = 12
# strlen('Olha a pamonha.') = 15
# strlen('Um, dois, feijão com arroz...') = 30

File: src/exercicios/05_EstruturasDeDados/04_Vetor/Strings/00-strlen.c
/**      @file: 00-strlen.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Implemente a função strlen. */

#include <stdio.h>

/* Retorna a quantidade de caracteres da string. */
int strlen(char* str) {
    /* Implemente o código aqui. */
}

int main() {
    char* str0 = NULL;
    char* str1 = "a";
    char* str2 = "Hello World!";
    char* str3 = "Olha a pamonha.";
    char* str4 = "Um, dois, feijão com arroz...";

    printf("strlen(\"%s\") = %d\n", str0, strlen(str0));
    printf("strlen(\"%s\") = %d\n", str1, strlen(str1));
    printf("strlen(\"%s\") = %d\n", str2, strlen(str2));
    printf("strlen(\"%s\") = %d\n", str3, strlen(str3));
    printf("strlen(\"%s\") = %d\n", str4, strlen(str4));

    return 0;
}

/* Resultado esperado:

strlen("(null)") = 0
strlen("a") = 1
strlen("Hello World!") = 12
strlen("Olha a pamonha.") = 15
strlen("Um, dois, feijão com arroz...") = 29 */
